src/ape/utils/trace.py

```diff
@@ -26,6 +40,22 @@
 _DEFAULT_INDENT = 2
 
 
+class TreeLike(Protocol):
```

Comment:
Wait, does protocol work with data model fields?

Reply:
Nope, it does not:

```py
>>> from typing import Protocol
>>> class TreeLike(Protocol):
...     a: int
...     b: bytes
>>> class Tree(TreeLike):
...     pass
>>> t = Tree()
>>> t.a
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 t.a

AttributeError: 'Tree' object has no attribute 'a'
```

Reply:
It is not really supposed to

Reply:
It doesn't work the way you're using it, in that a model doesn't inherit from the protocol, but I do think ultimately this needs a rework.
https://www.pythontutorial.net/python-oop/python-protocol/

Reply:
`evm-trace` is not going to import a base model from `ape`, nor really the other way around. 
`Protocol` is about typing / types / conformity.

I am open to other ideas but wanted to make sure the stage is set